services/venice-router/src/main/java/com/linkedin/venice/router/RouterServer.java

```diff
@@ -693,34 +693,35 @@ public boolean startInner() throws Exception {
     if (sslFactory.isPresent()) {
       sslInitializer = new SslInitializer(SslUtils.toAlpiniSSLFactory(sslFactory.get()), false);
       if (config.getClientSslHandshakeThreads() > 0) {
-        if (config.isResolveBeforeSSL()) {
-          ExecutorService sslHandshakeExecutor = registry.factory(ShutdownableExecutors.class)
-              .newFixedThreadPool(
-                  config.getClientSslHandshakeThreads(),
-                  new DefaultThreadFactory("RouterDNSBeforeSSLThread", true, Thread.NORM_PRIORITY));
-          int clientSslHandshakeThreads = config.getClientSslHandshakeThreads();
-          int maxConcurrentResolution = config.getMaxConcurrentResolutions();
-          int clientResolutionRetryAttempts = config.getClientResolutionRetryAttempts();
-          long clientResolutionRetryBackoffMs = config.getClientResolutionRetryBackoffMs();
-          if (useEpoll) {
-            sslResolverEventLoopGroup = new EpollEventLoopGroup(clientSslHandshakeThreads, sslHandshakeExecutor);
-          } else {
-            sslResolverEventLoopGroup = new NioEventLoopGroup(clientSslHandshakeThreads, sslHandshakeExecutor);
-          }
-          sslInitializer.enableResolveBeforeSSL(
-              sslResolverEventLoopGroup,
-              clientResolutionRetryAttempts,
-              clientResolutionRetryBackoffMs,
-              maxConcurrentResolution);
+        ThreadPoolExecutor sslHandshakeExecutor = ThreadPoolFactory.createThreadPool(
+            config.getClientSslHandshakeThreads(),
+            "SSLHandShakeThread",
+            config.getClientSslHandshakeQueueCapacity(),
+            LINKED_BLOCKING_QUEUE);
+        new ThreadPoolStats(metricsRepository, sslHandshakeExecutor, "ssl_handshake_thread_pool");
+        sslInitializer.enableSslTaskExecutor(sslHandshakeExecutor);
+      }
+      if (config.getResolveThreads() > 0) {
+        ThreadPoolExecutor dnsResolveExecutor = ThreadPoolFactory.createThreadPool(
+            config.getResolveThreads(),
+            "DNSResolveThread",
+            config.getResolveThreads(),
```

Comment:
This param is for queue length, so I suppose we should use a high number than the thread count.

Reply:
Good catch! Updated